File: 01-quadratic-equation/python/ecuacion_cuadratica.py
import math

class EcuacionCuadratica:
    """Clase que representa una ecuación cuadrática y resuelve sus raíces."""

    def __init__(self, a: float, b: float, c: float):
        self.a = a
        self.b = b
        self.c = c

    def resolver(self):
        """Resuelve la ecuación cuadrática y retorna sus raíces."""
        discriminante = self.b**2 - 4 * self.a * self.c
        if discriminante > 0:
            raiz_discriminante = math.sqrt(discriminante)
            x1 = (-self.b + raiz_discriminante) / (2 * self.a)
            x2 = (-self.b - raiz_discriminante) / (2 * self.a)
            return (x1, x2)
        elif discriminante == 0:
            x = -self.b / (2 * self.a)
            return (x,)
        else:
            return ()

class GestorEcuaciones:
    """Clase que gestiona múltiples ecuaciones cuadráticas y resuelve sus raíces."""

    def __init__(self):
        self.ecuaciones = []

    def agregar_ecuacion(self, ecuacion: EcuacionCuadratica):
        """Agrega una ecuación cuadrática al gestor."""
        self.ecuaciones.append(ecuacion)

    def resolver_ecuaciones(self):
        """Resuelve todas las ecuaciones almacenadas y muestra sus soluciones."""
        resultados = []
        for ecuacion in self.ecuaciones:
            soluciones = ecuacion.resolver()
            if len(soluciones) == 2:
                resultados.append(f"Raíces reales: {soluciones[0]} y {soluciones[1]}")
            elif len(soluciones) == 1:
                resultados.append(f"Una raíz real: {soluciones[0]}")
            else:
                resultados.append("No hay raíces reales.")
        return resultados

# Uso del gestor y ecuaciones
def ejecutar():
    gestor = GestorEcuaciones()
    gestor.agregar_ecuacion(EcuacionCuadratica(1, -3, 2))  # Raíces reales
    gestor.agregar_ecuacion(EcuacionCuadratica(1, 2, 1))   # Una raíz real
    gestor.agregar_ecuacion(EcuacionCuadratica(1, 0, 1))   # Sin raíces reales
    resultados = gestor.resolver_ecuaciones()
    for resultado in resultados:
        print(resultado)

# Ejecuta la función si el archivo se ejecuta directamente
ejecutar()

File: 01-quadratic-equation/java/EcuacionCuadratica.java
public class EcuacionCuadratica {
    private double a;
    private double b;
    private double c;

    public EcuacionCuadratica(double a, double b, double c) {
        this.a = a;
        this.b = b;
        this.c = c;
    }

    public double[] resolver() {
        double discriminante = b * b - 4 * a * c;

        if (discriminante > 0) {
            // Dos raíces reales
            double raiz1 = (-b + Math.sqrt(discriminante)) / (2 * a);
            double raiz2 = (-b - Math.sqrt(discriminante)) / (2 * a);
            return new double[] {raiz1, raiz2};
        } else if (discriminante == 0) {
            // Una raíz real
            double raiz = -b / (2 * a);
            return new double[] {raiz};
        } else {
            // No hay raíces reales
            return new double[] {};
        }
    }
}